cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/adapter/AdapterService.java

```diff
@@ -219,86 +213,101 @@ public void removeAdapter(String namespace, String adapterName) throws AdapterNo
+      Id.Program programId = Id.Program.from(namespaceId, spec.getName(), entry.getValue().getName());
+      addSchedule(programId, SchedulableProgramType.WORKFLOW, adapterSpec);
     }
   }
 
   // Unschedule all the programs needed for the adapter. Currently, only unscheduling of workflow is supported.
   private void unschedule(String namespaceId, ApplicationSpecification spec, AdapterTypeInfo adapterTypeInfo,
                           AdapterSpecification adapterSpec) {
-    ProgramType programType = adapterTypeInfo.getProgramType();
-    if (programType.equals(ProgramType.WORKFLOW)) {
-      Map<String, WorkflowSpecification> workflowSpecs = spec.getWorkflows();
-      for (Map.Entry<String, WorkflowSpecification> entry : workflowSpecs.entrySet()) {
-        Id.Program programId = Id.Program.from(namespaceId, adapterSpec.getType(), entry.getValue().getName());
-        deleteSchedule(programId, SchedulableProgramType.WORKFLOW,
-                       constructScheduleName(programId, adapterSpec.getName()));
-      }
-    } else {
-      // Only Workflows are supported to be unscheduled in the current implementation
-      throw new UnsupportedOperationException(String.format("Unsupported program type %s for adapter",
-                                                            programType.toString()));
+    checkWorkflow(adapterTypeInfo.getProgramType());
+    Map<String, WorkflowSpecification> workflowSpecs = spec.getWorkflows();
+    for (Map.Entry<String, WorkflowSpecification> entry : workflowSpecs.entrySet()) {
+      Id.Program programId = Id.Program.from(namespaceId, adapterSpec.getType(), entry.getValue().getName());
+      deleteSchedule(programId, SchedulableProgramType.WORKFLOW,
+                     constructScheduleName(programId, adapterSpec.getName()));
     }
   }
 
+  private void checkWorkflow(ProgramType programType) {
+    // Only Workflows are supported to be scheduled in the current implementation
+    Preconditions.checkArgument(programType.equals(ProgramType.WORKFLOW),
+                                String.format("Unsupported program type %s for adapter", programType.toString()));
+  }
+
   // Adds a schedule to the scheduler as well as to the appspec
-  private void addSchedule(Id.Program programId, AdapterSpecification adapterSpec) {
-    String cronExpr = Schedules.toCronExpr(adapterSpec.getProperties().get("frequency"));
-    Preconditions.checkNotNull(cronExpr, "Frequency of running the adapter is missing. Cannot schedule program");
+  private void addSchedule(Id.Program programId, SchedulableProgramType programType, AdapterSpecification adapterSpec) {
+    String frequency = adapterSpec.getProperties().get("frequency");
+    Preconditions.checkArgument(frequency != null,
+                                "Frequency of running the adapter is missing from adapter properties." +
+                                  " Cannot schedule program.");
```

Comment:
Intellij automatically indents it slightly, if the bottom line is a part of the same String, and not a new parameter.